src/components/NetworkAssetSelectInput/NetworkAssetSelectInput.tsx

```diff
@@ -305,11 +314,48 @@ const NetworkAssetSelectInput = ({
     walletAddress,
   ]);
 
+  useEffect(() => {
+    const handleBalanceGet = async () => {
+      if (!sdk || !walletAddress) return;
+      await getSmartWalletBalancesByChain(walletAddress, supportedChains);
+    };
+    handleBalanceGet();
+  }, [supportedChains, sdk, walletAddress]);
+
   const filteredSelectedNetworkAssets = useMemo(() => {
     const filtered = selectedNetworkAssets.filter((asset) => containsText(asset.name, assetSearchQuery) || containsText(asset.symbol, assetSearchQuery));
     return orderBy(filtered, [(asset) => asset.balanceWorthUsd ?? 0, 'name'], ['desc', 'asc']);
   }, [selectedNetworkAssets, assetSearchQuery]);
 
+  useEffect(() => {
+    const updateAvalancheBalance = () => {
+      if (!sdk ||
+        !walletAddress ||
+        !preselectedNetwork ||
+        preselectedNetwork.chainId !== CHAIN_ID.AVALANCHE ||
+        !smartWalletBalanceByChain ||
+        filteredSelectedNetworkAssets.length
+      ) {
+        return;
+      }
+      setsmartWalletBalanceByChain([
+        ...smartWalletBalanceByChain,
```

Comment:
Why not reusing current state from `setsmartWalletBalanceByChain((current) => ...` ?

Reply:
done with PRO-640